include/bluetooth/bluetooth.h

```diff
@@ -279,6 +279,21 @@ int bt_le_scan_start(const struct bt_le_scan_param *param, bt_le_scan_cb_t cb);
  */
 int bt_le_scan_stop(void);
 
+/** @brief Helper for parsing advertising (or EIR) data.
+ *
+ *  A helper for parsing the advertsing data received in the callback
+ *  given to bt_le_scan_start().
+ *
+ *  @param ad        Advertising data as given to the bt_le_scan_cb_t callback.
+ *  @param func      Callback function which will be called for each element
+ *                   that's found in the data. The callback should return
+ *                   true to continue parsing, or false to stop parsing.
+ *  @param user_data User data to be passed to the callback.
+ */
+void bt_data_parse(struct net_buf_simple *ad,
```

Comment:
I was gonna suggest a new name but I see that the CSS calls this "Data Types" so this is fine IMHO